lib/lightning/utils/maps.ex

```diff
@@ -13,4 +26,44 @@ defmodule Lightning.Utils.Maps do
         {key, value}
     end)
   end
+
+  @doc """
+  Recursively converts all map keys from atoms to strings throughout a nested structure.
+
+  Handles:
+  - Nested maps
+  - Lists of maps
+  - Structs (converted to maps first)
+
+  Note: Does not perform nil key checks like `stringify_keys/1` for performance reasons.
+
+  ## Examples
+
+      iex> deep_stringify_keys(%{user: %{name: "John", roles: [:admin]}})
+      %{"user" => %{"name" => "John", "roles" => [:admin]}}
+
+      iex> deep_stringify_keys(%{items: [%{id: 1}, %{id: 2}]})
+      %{"items" => [%{"id" => 1}, %{"id" => 2}]}
+
+      iex> deep_stringify_keys(%User{name: "John", email: "john@example.com"})
+      %{"name" => "John", "email" => "john@example.com"}
+  """
+  def deep_stringify_keys(%_{} = struct) do
```

Comment:
@elias-ba It feels like we could write some nice clean tests to cover this functionality?

Reply:
I removed this func. It's no longer needed. Changed my approach